pkg/accounting/tally/stats.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package tally
+
+import (
+	"storj.io/storj/pkg/pb"
+)
+
+//Stats all stats fields
+type Stats struct {
+	Segments        int64
+	InlineSegments  int64
+	RemoteSegments  int64
+	UnknownSegments int64
+
+	Files       int64
+	InlineFiles int64
+	RemoteFiles int64
+
+	Bytes       int64
+	InlineBytes int64
+	RemoteBytes int64
+}
+
+// Combine aggregates all the stats
+func (s *Stats) Combine(o *Stats) {
+	s.Segments += o.Segments
+	s.InlineSegments += o.InlineSegments
+	s.RemoteSegments += o.RemoteSegments
+	s.UnknownSegments += o.UnknownSegments
+
+	s.Files += o.Files
+	s.InlineFiles += o.InlineFiles
+	s.RemoteFiles += o.RemoteFiles
+
+	s.Bytes += o.Bytes
+	s.InlineBytes += o.InlineBytes
+	s.RemoteBytes += o.RemoteBytes
+}
+
+// AddSegment groups all the data based the passed pointer
+func (s *Stats) AddSegment(pointer *pb.Pointer, last bool) {
+	s.Segments++
+	switch pointer.GetType() {
+	case pb.Pointer_INLINE:
+		s.InlineSegments++
+		s.InlineBytes += int64(len(pointer.InlineSegment))
+		s.Bytes += int64(len(pointer.InlineSegment))
+
+	case pb.Pointer_REMOTE:
+		s.RemoteSegments++
+		s.RemoteBytes += pointer.GetSegmentSize()
+		s.Bytes += pointer.GetSegmentSize()
+	default:
+		s.UnknownSegments++
+	}
+
+	if last {
+		s.Files++
+		switch pointer.GetType() {
+		case pb.Pointer_INLINE:
+			s.InlineFiles++
+		case pb.Pointer_REMOTE:
+			s.RemoteFiles++
+		}
+	}
+}
+
+// Report reports the stats thru monkit
+func (s *Stats) Report(prefix string) {
```

Comment:
Add the dot inside here instead of supplying it from outside. Less likely to make a mistake.

Reply:
Callers should use `Report("bucket")` instead of `Report("bucket.")`. Inside the func use `mon.IntVal(prefix + ".segments").Observe(s.Segments)`.